noise/README.md

```diff
@@ -204,54 +204,86 @@ the early data MUST be included as described below.
 
 #### The libp2p Signed Handshake Payload
 
-libp2p-specific data, including the signature used for static key
-authentication, is transmitted in Noise handshake message payloads. When
-decrypted, the message payload has the structure described in [Encrypted
+libp2p-specific early data, plus a signature that links the static Noise key
+of the session with the libp2p identity key (for authentication), is
+transmitted in Noise handshake message payloads.
+
+These payloads MUST appear in the first handshake message that guarantees
+secrecy, sender authentication, and integrity:
+
+* In XX-initiated handshakes, the initiator will send its payload in message 3
+  (closing message), whereas the responder will send it in message 2 (its only
+  message).
+* In IK-initiated optimistic handshakes, the initiator will send its message
+  payload message 1, as it satisfies the guarantees.
+    * If the responder continues the IK handshake, it will send its message
+      payload in message 2.
+    * If the responder fall backs to XXfallback, it won't be able to decrypt
+      the initiator's message data from message 1. Therefore, the initiator
+      MUST rebuild the message payload with the new cryptographic material,
+      and resend it in the 3rd message.
+
+To guard against replay attacks, these message payloads are signed with the
```

Comment:
Shouldn't you be signing the[ Handhash](https://github.com/noiseprotocol/noise_spec/blob/master/noise.md#112-channel-binding) + the message payload to bind the signature to a specific channel.

Signing with an X25519 keys can get annoying. Signal protocol does it but X25519 to ed25519 is not provided by many implementations.

Reply:
Yeah easier to make the static keys Ed25519 if you want this.